x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/RemoteAccessAuthenticationService.java

```diff
@@ -150,7 +150,19 @@ public AuthenticationService getAuthenticationService() {
     }
 
     private void validate(final RemoteAccessAuthentication remoteAccessAuthentication) {
-        final Subject effectiveSubject = remoteAccessAuthentication.getAuthentication().getEffectiveSubject();
+        final Authentication authentication = remoteAccessAuthentication.getAuthentication();
+        authentication.checkInternalConsistency();
```

Comment:
This is the only place where the consistency check is enforced in production code. I intentionally keep the check as assertion for existing code and not run them in production because it is (1) not necessary when authentication is built internally and (2) avoid any accidental changes to existing logic.